packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts

```diff
@@ -0,0 +1,355 @@
+        getTestNode(parentDataStore.id).children.delete(dataStore.id);
+        getTestNode(dataStore.id).parents.delete(parentDataStore.id);
+        updateUnreferencedDataStores(dataStore.id);
+    };
+
+    const getRandomContainer = (): IContainer => {
+        const id = randomInteger(0, connectedContainers.length - 1);
+        return connectedContainers[id];
+    };
+
+    const getRandomDataStoreFromContainer = async (container: IContainer) => {
+        const random = randomInteger(0, testNodes.size - 1);
+        const dataStoreId = Array.from(testNodes.values())[random].id;
+        return requestFluidObject<TestDataObject>(container, dataStoreId);
+    };
+
+    const getRandomParentForDataStore = async (dataStore: TestDataObject, container: IContainer) => {
+        const parents = getTestNode(dataStore.id).parents;
+        const random = randomInteger(0, parents.size - 1);
+        const parentId = Array.from(parents.values())[random];
+        return requestFluidObject<TestDataObject>(container, parentId);
+    };
+
+    const referenceRandomDataStore = async () => {
+        if (connectedContainers.length <= 0 || testNodes.size <= 0) {
+            return;
+        }
+        const container = getRandomContainer();
+        const dataStore = await getRandomDataStoreFromContainer(container);
+        const parent = await getRandomDataStoreFromContainer(container);
+        referenceDataStore(dataStore, parent);
+    };
+
+    const unreferenceRandomDataStore = async () => {
+        if (connectedContainers.length <= 0 || testNodes.size <= 0) {
+            return;
+        }
+        const container = getRandomContainer();
+        const dataStore = await getRandomDataStoreFromContainer(container);
+        const parentDataStore = await getRandomParentForDataStore(dataStore, container);
+        unreferenceDataStore(dataStore, parentDataStore);
+    };
+
+    const actionsList: (() => Promise<unknown>)[] = [
+        loadNewContainer,
+        closeRandomContainer,
+        createDataStoreForRandomContainer,
+        referenceRandomDataStore,
+        unreferenceRandomDataStore,
+    ];
+    let testTime = 0;
+    const testEnd = 60 * 1000; // 1 minute
+    const testActions: { sleep: number; action: () => Promise<unknown>; }[] = [];
+    while (testTime < testEnd) {
+        const sleepTime: number = randomInteger(0, defaultSessionExpiryDurationMs + 1000);
+        const action = makeRandom().pick(actionsList);
+        testActions.push({ sleep: sleepTime, action });
+        testTime += sleepTime;
+    }
+
+    it("Create and reference and unreference datastores with multiple containers", async () => {
+        overrideLogger.ignoreExpectedEventTypes({
+            eventName: "fluid:telemetry:Container:ContainerClose",
+            errorType: ContainerErrorType.clientSessionExpiredError,
+        });
+
+        for (const action of testActions) {
+            void action.action();
```

Comment:
`void` is hiding the fact that this is a floating promise.  I think you should await it so if it rejects (any assert or other error thrown in the action) the test fails.  If you want the sleep to run in parallel, then do `await Promise.all(...)`.  Also, as-is there's an interesting race condition where two subsequent actions could theoretically run simultaneously if the first one is really slow for some reason and sleep is really short.  Not sure we want to be testing the race conditions there (your TestNode code looks like it's intended to run synchronously)